spec/index.bs

```diff
@@ -3765,17 +3765,18 @@ interface GPUQueue {
     void signal(GPUFence fence, GPUFenceValue signalValue);
 
     void writeBuffer(
-        ArrayBuffer sourceData,
-        GPUSize64 sourceOffset,
-        GPUBuffer destination,
-        GPUSize64 destinationOffset,
-        GPUSize64 size);
+        GPUBuffer buffer,
+        GPUSize64 bufferOffset,
+        ArrayBuffer data,
+        optional GPUSize64 dataOffset = 0,
+        optional GPUSize64 size = 0);
 
     void writeTexture(
-      ArrayBuffer sourceData,
-      GPUTextureDataLayout sourceLayout,
       GPUTextureCopyView destination,
-      GPUExtent3D size);
+      GPUExtent3D size,
+      GPUTextureDataLayout dataLayout,
+      ArrayBuffer data,
+      optional GPUSize64 dataOffset = 0);
```

Comment:
Maybe we should just make a new `GPUArrayBufferCopyView : GPUTextureDataLayout` so this can be nice and uniform with the rest? `(destination, source, size)` or `(source, destination, size)`.

Reply:
IMHO writes aren't like copies and don't need to match. What users should see is should be closest to `resource.write(stuff, arraybuffer)`.

Reply:
Fine with me